Pricing page: Merge products and bundles section into one

jkguidaven opened the pull request:
#### Proposed Changes

* This PR will merge the Products and Bundles sections into one.
<img width="443" alt="Screen Shot 2022-12-19 at 12 32 22 PM" src="https://user-images.githubusercontent.com/56598660/208350325-36898be3-76c0-47b6-b5e5-c4d826b2b6cb.png">


#### Testing Instructions
1. Run this branch by following the steps below (or you can use the Jetpack Cloud live link commented on this PR)
    * Run `git fetch && git checkout update/merge-pricing-page-product-bundle`
    * Run `yarn start-jetpack-cloud`

2. Goto http://jetpack.cloud.localhost:3000/pricing or use the Jetpack cloud live link below and append `/pricing`
3. Confirm that the pricing page now has the new design

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to 1202858161751496-as-1203568228122298

jkguidaven committed:
Remove pricing page product/bundle toggle.

jkguidaven committed:
Merge product and bundle tabs.

jkguidaven committed:
Fix margin of item list component to match new design.

jkguidaven committed:
Rename class name and variable for better context.

manzoorwanijk commented on client/my-sites/plans/jetpack-plans/product-store/items-list/style-most-popular.scss:
```diff
@@ -72,3 +72,30 @@
+		align-items: center;
+		gap: 4px;
+	}
+
```
Not related to this PR, but I see a slight misalignment of the info button for the amount saved.

Can we fix that with this bit of styling here?

```suggestion

	.amount-saved--text button {
		display: flex;
	}

```

BEFORE
<img width="1225" alt="Screenshot 2022-12-19 at 11 49 02 AM" src="https://user-images.githubusercontent.com/18226415/208361075-3a1607a5-7a7e-4c83-a380-650b233d6fa0.png">

AFTER
<img width="1231" alt="Screenshot 2022-12-19 at 11 49 53 AM" src="https://user-images.githubusercontent.com/18226415/208361098-fce61d7c-e084-48e2-9d62-18785c740df2.png">

CC: @ilonagl

ilonagl replied:
Great catch!

manzoorwanijk reviewed (approved):
This looks good and works as expected. I have a minor suggestion for improvement, not related to this PR.

jkguidaven committed:
Fix misalignment of the info button in the amount saved component.

Co-authored-by: Manzoor Wani <manzoorwani.jk@gmail.com>

ilonagl commented:
![Spacing-2](https://user-images.githubusercontent.com/100145243/208413536-78e8b635-d2e8-41a5-a7b6-b48222e897b5.png)
![Spacing-3](https://user-images.githubusercontent.com/100145243/208413546-0c6a8b13-1ce0-4068-9872-02f070a46764.png)


@jkguidaven It looks like spacing is a bit off on mobile. I added some screenshots with spacing marks above. Please take a look and let me know your thoughts!

jkguidaven committed:
Fix spacing in mobile view.

dereksmart commented:
Not sure if a regression - maybe with the [browser history stuff](https://github.com/Automattic/wp-calypso/pull/71300/files#diff-04270c5c1e3d5f1a470b5d0d29c71f9a99c46b83cd5e5453c8845dab62ea982fL87) being removed, or if it's a calypso.live thing: "Remove from cart" takes me back to https://wordpress.com/start/domains, rather than cloud.jetpack.com/pricing as production.

jkguidaven commented:
> Not sure if a regression - maybe with the [browser history stuff](https://github.com/Automattic/wp-calypso/pull/71300/files#diff-04270c5c1e3d5f1a470b5d0d29c71f9a99c46b83cd5e5453c8845dab62ea982fL87) being removed, or if it's a calypso.live thing: "Remove from cart" takes me back to https://wordpress.com/start/domains, rather than cloud.jetpack.com/pricing as production.
> 
> Not a blocker for merging these changes. The rest looks good. We can address it in another PR if it's indeed regression.


Thanks for catching this. I didn't realize this issue existed until you mentioned it. It looks like it is not related to the changes made here. I can also reproduce the issue in other PRs' live links (e.g., https://container-elastic-kapitsa.calypso.live/pricing). The good thing is that the issue is not reproducible when running Calypso locally, so it is most likely an issue exclusive to the live links but I will check after merging this PR incase it resurface.

jkguidaven commented:
> @jkguidaven It looks like spacing is a bit off on mobile. I added some screenshots with spacing marks above. Please take a look and let me know your thoughts!

Thanks for reviewing this. I have fixed the spacing in the mobile view as you have pointed out.
